src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_Scope.cs

```diff
@@ -13787,5 +13787,379 @@ static void Main(int i)
+                // (24,9): error CS0029: Cannot implicitly convert type 'bool' to 'int'
+                //         int[true is var x3, x3] _5;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "int[true is var x3, x3]").WithArguments("bool", "int").WithLocation(24, 9),
+                // (24,12): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         int[true is var x3, x3] _5;
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[true is var x3, x3]").WithLocation(24, 12),
+                // (24,25): error CS0136: A local or parameter named 'x3' cannot be declared in this scope because that name is used in an enclosing local scope to define a local or parameter
+                //         int[true is var x3, x3] _5;
+                Diagnostic(ErrorCode.ERR_LocalIllegallyOverrides, "x3").WithArguments("x3").WithLocation(24, 25),
+                // (24,33): warning CS0168: The variable '_5' is declared but never used
+                //         int[true is var x3, x3] _5;
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "_5").WithArguments("_5").WithLocation(24, 33),
+                // (30,12): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         int[x4] _6;
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[x4]").WithLocation(30, 12),
+                // (30,17): warning CS0168: The variable '_6' is declared but never used
+                //         int[x4] _6;
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "_6").WithArguments("_6").WithLocation(30, 17),
+                // (31,12): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         int[true is var x4, x4] _7;
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[true is var x4, x4]").WithLocation(31, 12),
+                // (31,25): error CS0128: A local variable or function named 'x4' is already defined in this scope
+                //         int[true is var x4, x4] _7;
+                Diagnostic(ErrorCode.ERR_LocalDuplicate, "x4").WithArguments("x4").WithLocation(31, 25),
+                // (31,33): warning CS0168: The variable '_7' is declared but never used
+                //         int[true is var x4, x4] _7;
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "_7").WithArguments("_7").WithLocation(31, 33),
+                // (36,9): error CS0029: Cannot implicitly convert type 'bool' to 'int'
+                //         int[true is var x5, x5] _8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "int[true is var x5, x5]").WithArguments("bool", "int").WithLocation(36, 9),
+                // (36,9): error CS0029: Cannot implicitly convert type 'bool' to 'int'
```

Comment:
>// (36,9): error CS0029: Cannot implicitly convert type 'bool' to 'int' [](start = 16, length = 71)

I assume this error and the previous error are about different expressions, but for some reason both have the same location - the entire type syntax. I think we should use more precise location.

Reply:
This is the current behavior on master - see IOperationTests_ArrayCreationAndInitializer.ArrayCreationErrorCase_MissingExplicitCast for example:
> Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "new string[b]").WithArguments("object", "int").WithLocation(6, 27)
This is fine for that scenario, as only the one expression will be bound. However, it causes the issue you noticed above whenever we have two such errors.
Try this in master for example:  `int[,] x = new int[true, true];`
Also notice the location of the red squigglies here:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAGAAhwEYBuAWACgcBmIgJgIGECBvKgzounAFgICyACgCUbDl0loAdjADaSALoEAHgQC8BaXADuBGfJgIArnCQEjpxRUqSAvlTtA=

I think that you are right therefore that we should just use the dimension, and so I will change this.